WMF Framework/Router.swift

```diff
@@ -0,0 +1,162 @@
+@objc(WMFRouter)
+public class Router: NSObject {
+    public enum Destination: Equatable {
+        case inAppLink(_: URL)
+        case externalLink(_: URL)
+        case article(_: URL)
+        case articleHistory(_: URL, articleTitle: String)
+        case articleDiffCompare(_: URL, fromRevID: Int?, toRevID: Int?)
+        case articleDiffSingle(_: URL, fromRevID: Int?, toRevID: Int?)
+        case userTalk(_: URL)
+        case search(_: URL, term: String?)
+    }
+    
+    unowned let configuration: Configuration
+    required init(configuration: Configuration) {
+        self.configuration = configuration
+    }
+    
+    private let mobilediffRegexCompare = try! NSRegularExpression(pattern: "^mobilediff/([0-9]+)\\.\\.\\.([0-9]+)", options: .caseInsensitive)
```

Comment:
Could maybe use Swift 5 raw strings ( https://www.hackingwithswift.com/articles/162/how-to-use-raw-strings-in-swift ) now so we don't have to do escaping?